### PR TITLE
Not indexing large text fields

### PR DESCRIPTION
**Pull request by r0goyal**

### CODE REVIEW

File: foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/mutator/LargeTextNodeRemover.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
+package com.flipkart.foxtrot.core.querystore.mutator;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+public class LargeTextNodeRemover implements IndexerEventMutator {
+
+    private static final int MAX_SIZE = 500;
```
[the remark this answers is not in the document]
Comment (r0goyal) [reply foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/mutator/LargeTextNodeRemover.java]: More than 1 KB text fields ?

File: foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/mutator/LargeTextNodeRemover.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                        handleArrayNode((ArrayNode) node);
+                    } else if (node.isTextual()) {
+                        copyNode = node.isNull() || node.textValue().length() <= MAX_SIZE;
+                    }
+                    if (copyNode) {
+                        copy.add(node);
+                    }
+                });
+        arrayNode.removeAll();
+        arrayNode.addAll(copy);
```
Comment (r0goyal) [inline foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/mutator/LargeTextNodeRemover.java]: @santanusinha @nitishgoyal13 Any efficient way of doing this ?

Comment (santanusinha) [reply foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/mutator/LargeTextNodeRemover.java]: Nope

File: foxtrot-core/src/main/java/com/flipkart/foxtrot/core/config/IndexConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class IndexConfiguration {
+
+    @Builder.Default
+    private int textFieldMaxSize = 1000;
```
Comment (santanusinha) [inline foxtrot-core/src/main/java/com/flipkart/foxtrot/core/config/IndexConfiguration.java]: maxIndexedTextFieldSize

File: foxtrot-core/src/test/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStoreTest.java [hunk 4]
```diff
@@ -151,6 +153,30 @@ public void testSaveSingleRawKeyVersion2() throws Exception {
...
+                .setStoredFields(ElasticsearchUtils.DOCUMENT_META_TIMESTAMP_FIELD_NAME, "testField", "testLargeField")
+                .execute()
+                .actionGet();
+        System.out.println(getResponse.getFields());
+        assertTrue("Id should exist in ES", getResponse.isExists());
+        assertEquals("Id should match requestId", originalDocument.getId(), getResponse.getId());
+        assertNull(getResponse.getField("testLargeField"));
+        assertNotNull(getResponse.getField("testField"));
+    }
+
```
Comment (santanusinha) [inline foxtrot-core/src/test/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStoreTest.java]: Add a perf test if possible

File: foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/mutator/LargeTextNodeRemover.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        }
+    }
+
+    private void handleObjectNode(ObjectNode objectNode) {
+        if (objectNode == null || objectNode.isNull()) {
+            return;
+        }
+        objectNode.fields().forEachRemaining(entry -> {
+            if (entry.getValue().isTextual()) {
+                if (entry.getValue().size() > MAX_SIZE) {
```
[the remark this answers is not in the document]
Comment (r0goyal) [reply foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/mutator/LargeTextNodeRemover.java]: Done

File: foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/mutator/LargeTextNodeRemover.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (r0goyal) [reply foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/mutator/LargeTextNodeRemover.java]: Made it configurable

**Comment by r0goyal** [comment]
@santanusinha Test cases added. Please review again

**Pull request merged**